### PR TITLE
feat(get-notifications): add use case implementation

### LINKED ISSUE #19: Use case: Getting all notifications related with a note.
This use case will receive a note identifier and return a collection of notifications associated with this note.
Be aware of the following cases:
- The note doesn't have any notification.
- The note has only one notification associated.
- The note has multiple notifications associated.

### PR DESCRIPTION
**Pull request by FerranAD**
This closes #19.

### CODE REVIEW

File: src/main/kotlin/notification/domain/Notification.kt [hunk 1]
```diff
@@ -2,4 +2,4 @@ package notification.domain
 
 import shared.domain.Identifier
 
-class Notification(val id: Identifier)
+class Notification(val noteId: Identifier)
```
Comment (Pablito2020) [inline src/main/kotlin/notification/domain/Notification.kt:5]: I suppose that you didn't add the Date member because you're doing TDD and, in this use case, isn't needed. If it's the case well done! 😺

File: src/test/kotlin/notification/application/NotificationGetterTest.kt [hunk 2]
```diff
@@ -1,45 +1,60 @@
...
 
+import notification.domain.Notification
 import notification.domain.NotificationRepository
 import notification.mothers.NotificationMother
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.mockito.Mockito
 import shared.domain.IdentifierGenerator
 import shared.mothers.IdentifierMother
+import java.util.*
```
Comment (Pablito2020) [inline src/test/kotlin/notification/application/NotificationGetterTest.kt:11]: Why is this needed? Aren't we using the kotlin collections API?

Comment (FerranAD) [reply src/test/kotlin/notification/application/NotificationGetterTest.kt:11]: Probably an unused import that I forgot to remove, I'll check.

**Comment by oriolagobat** [comment]
As

**Review by oriolagobat — DISMISSED** [review]

File: src/main/kotlin/notification/application/NotificationSearcher.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class NotificationSearcher(
+    private val repository: NotificationRepository,
+) {
+    /***
+     * Looks for all Notifications with the given Note identifier in the repository.
+     * @see Notification
+     * @see Note
+     * @return A collection of Notification instances (empty if none found)
+     */
+    fun search(noteId: String) = search(Identifier(noteId))
```
Comment (oriolagobat) [inline src/main/kotlin/notification/application/NotificationSearcher.kt:17]: Creating an `Identifier` from an invalid string throws `InvalidUUIDException`.
The `KDoc` should relfect that.

**Review by oriolagobat — CHANGES_REQUESTED** [review]
A missing thrown exception on the `KDoc`

**Review by Pablito2020 — DISMISSED** [review]
Be aware of what @oriolagobat told you about the docs, once you do his requested change this feature is ready to be merged.

**Review by oriolagobat — APPROVED** [review]

**Review by Pablito2020 — APPROVED** [review]
